providers/silverstripe.go

```diff
@@ -0,0 +1,44 @@
+package providers
+
+import (
+	"fmt"
+	"log"
+	"net/http"
+
+	"github.com/bitly/oauth2_proxy/api"
+)
+
+type SilverStripeProvider struct {
+	*ProviderData
+}
+
+func NewSilverStripeProvider(p *ProviderData) *SilverStripeProvider {
+	p.ProviderName = "SilverStripe"
+	if p.Scope == "" {
+		p.Scope = "read_profile"
+	}
+	return &SilverStripeProvider{ProviderData: p}
+}
+
+func (p *SilverStripeProvider) getProfileField(s *SessionState, field string) (string, error) {
+	req, err := http.NewRequest("GET", p.ValidateURL.String(), nil)
+	req.Header.Set("Authorization", fmt.Sprintf("Bearer %s", s.AccessToken))
+	if err != nil {
+		log.Printf("failed building request %s", err)
+		return "", err
+	}
+	json, err := api.Request(req)
+	if err != nil {
+		log.Printf("failed making request %s", err)
+		return "", err
+	}
+	return json.Get(field).String()
```

Comment:
Can we cache this inside the `*p`? It's a bit silly we are making two calls where we could be making one.